<launch>
    <!-- args --> 
    <!-- Behzad's workstation ip --> 
    <!-- arg name="host_ip" value="10.157.34.101" /--> 
    <arg name="host_ip" default="$(env host_ip)" />
    <!--arg name="host_ip" value="169.254.153.158" /--> 
    <!-- Hasans's workstation ip --> 
    <!-- arg name="host_ip" value="10.157.34.208" /--> 
    <arg name="img_width" value="512"/>
    <arg name="img_height" value="288"/>
    <arg name="CLCT_DATA" value="true" />

    <!-- orb slam location--> 

    <!-- Set the localization method -->
    <!-- arg name="localization_method" value="orb_slam2_rgbd" /-->
    <!-- arg name="localization_method" value="ground_truth" /-->
    <arg name="localization_method" value="gps" />
    <param name="localization_method" value="$(arg localization_method)" />
    <param name="detec_fail_ctr_threshold" value="10" />
    <param name="supervisor_mailbox" value="$(find package_delivery)/../data/follow_the_leader/supervisor_mailbox.txt" />
    <param name="stats_file_addr" value="$(find package_delivery)/../data/follow_the_leader/stats.json" />

    
    <param name="image_w__global" value="$(arg img_width)"/>
    <param name="image_h__global" value="$(arg img_height)"/>
    <param name="height_ratio" value=".25"/> 
    
    <!-- setting DEBUG to false prevents publishing and subscribing to rviz /--> 
    <!-- also planning to prevent stdio stuff(if they really matter)--> 
    <arg name="DEBUG" value="true" /> 
    <param name="ros_DEBUG" type="bool" value="$(arg DEBUG)" />
    <param name="frame_to_process_upper_bound" type="int" value="2"/> 
    
    <!-- sensors /-->
    <arg name="all_front" default="true"/>	
    <param name="loop_rate" value="150"/>
    
    <comment>
    <!--img publisher node--> 
    <node pkg="airsim_img_publisher" type="airsim_imgPublisher" name="airsim_imgPublisher" output="screen">
    <!-- IP of the Airsim computer -->
    <param name="Airsim_ip" value="$(arg host_ip)" />
    
    <!-- Port of the Airsim computer (this is default, do not change!) -->
    <param name="Airsim_port" value="41451"/>

    <!-- RGB camera parameters (parameters when calibrated) -->
    <comment> 
    <param name="Fx" value="128"/>
    <param name="Fy" value="128"/>
    <param name="cx" value="128"/>
    <param name="cy" value="72"/>
    <param name="scale_x" value="256"/>  <!-- Width when calibrated!-->
    <param name="scale_y" value="144"/> <!-- Height when calibrated!-->
    <param name="width" value="256"/>  <!-- Width when calibrated!-->
    <param name="height" value="144"/> <!-- Height when calibrated!-->
    </comment> 
    <param name="Fx" value="$(eval arg('img_width')/2)"/>
    <param name="Fy" value="$(eval arg('img_width')/2)"/>
    <param name="cx" value="$(eval arg('img_width')/2)"/>
    <param name="cy" value="$(eval arg('img_height')/2)"/>
    <param name="scale_x" value="$(arg img_width)"/>  <!-- Width when calibrated!-->
    <param name="scale_y" value="$(arg img_height)"/> <!-- Height when calibrated!-->
    <param name="width" value="$(arg img_width)"/>  <!-- Width when calibrated!-->
    <param name="height" value="$(arg img_height)"/> <!-- Height when calibrated!-->
    <!--/comment-->

    <param name="maxDist" value="50.0"/> 
    <!-- Depth baseline -->
    <param name="Tx" value="0.14"/>
    <param name="scale" value="0.38"/> 

    <!-- Topic with odometry information for the vehicle (used to create tf tree) -->
    <!-- we are not using the odomTopic since arisim is not providing it at themoment -->	
    <param name="odomTopic" value="/mavros/local_position/odom"/>
    
    <!-- Localization of the depth images -->
    <param name="localization_method" value="$(arg localization_method)" />

    </node>
    </comment> 
    
    
    <!-- img_publisher point cloud /-->
    <node pkg="airsim_img_publisher" type="airsim_imgPublisher" name="airsim_imgPublisher" output="screen">

    <!-- IP of the Airsim computer -->
    <param name="all_front" value="$(arg all_front)"/>	
    <param name="localization_method" value="$(arg localization_method)" />
    <param name="Airsim_ip" value="$(arg host_ip)" />

    <param name="loop_rate" value="20" />

    <!-- Port of the Airsim computer (this is default, do not change!) -->
    <param name="Airsim_port" value="41451"/>

    <!-- RGB camera parameters (parameters when calibrated) -->
    <param name="Fx" value="256"/>
    <param name="Fy" value="256"/>
    <param name="cx" value="256"/>
    <param name="cy" value="144"/>
    <param name="scale_x" value="512"/>  <!-- Width when calibrated!-->
    <param name="scale_y" value="288"/> <!-- Height when calibrated!-->
    <param name="width" value="512"/>  <!-- Width when calibrated!-->
    <param name="height" value="288"/> <!-- Height when calibrated!-->

    <comment> 
    <param name="Fx" value="200"/>
    <param name="Fy" value="200"/>
    <param name="cx" value="200"/>
    <param name="cy" value="150"/>
    <param name="scale_x" value="400"/>  <!-- Width when calibrated!-->
    <param name="scale_y" value="300"/> <!-- Height when calibrated!-->
    <param name="width" value="400"/>  <!-- Width when calibrated!-->
    <param name="height" value="300"/> <!-- Height when calibrated!-->
    </comment>

    <param name="maxDist" value="50.0"/> 
    <!-- Depth baseline -->
    <param name="Tx" value="0.14"/>
    <param name="scale" value="0.38"/> 

    <!-- Topic with odometry information for the vehicle (used to create tf tree) -->
    <!-- we are not using the odomTopic since arisim is not providing it at themoment -->	
    <param name="odomTopic" value="/mavros/local_position/odom"/>
  </node>




    <include file="$(find follow_the_leader)/launch/detection_node.launch"/>
    <include file="$(find follow_the_leader)/launch/tracking_node.launch"/>

    <param name="ip_addr" type="string" value="$(arg host_ip)" />
    <node pkg="follow_the_leader" type="follow_the_leader_node" name="follow_the_leader_node"  output="screen"/>
    <!-- param name="mav_name" type="string" value="$(arg mav_name)" / -->
    <comment launch-prefix="gdb -ex run --args"/>


    <node pkg="follow_the_leader" type="pid_node" name="pid_node" output="screen">
        <!--param name="vx__P__global" value=".08"/-->
        <param name="vx__P__global" value=".03"/>
        
        <param name="vy__P__global" value="5"/>

        <param name="vz__P__global" value=".03"/>

        <param name="vx__I__global" value=".00"/>
        <param name="vy__I__global" value=".00"/>
        <param name="vz__I__global" value=".000"/>

        <param name="vx__D__global" value=".00"/>
        <param name="vy__D__global" value=".00"/>
        <param name="vz__D__global" value=".00"/>

        <param name="ip_addr__global" value="$(arg host_ip)"/>
    </node>

    <node pkg="profile_manager" type="profile_manager" name="profile_manager" output="screen">
    <param name="ip_addr" value="$(arg host_ip)"/>
    </node>

    <group if="$(arg CLCT_DATA)">
    <param name="enable_statistics" value="true"/>
    </group> 

</launch>
